src/Todo/TodoAppModel.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import AddEditTodo from "../components/AddEditTodo";
+import TodoHead from "./TodoHead";
+import TodoList from "../Todo/TodoList";
+import { setTodos } from "../redux/todoSlice";
+import { toggleAddTodoVisibility } from "../redux/uiSlice";
+
+function TodoAppModel() {
+    const dispatch = useDispatch();
+    const todos = useSelector((state) => state.todos.todos);
+    const { isAddTodoVisible } = useSelector((state) => state.ui);
+
+    useEffect(() => {
+        const savedTodos = JSON.parse(localStorage.getItem("todos"));
```

Comment:
this set todos should be done inside redux initial state too not in component @kaushaki-kumari